aws_lambda_powertools/event_handler/util.py

```diff
@@ -40,3 +69,30 @@ def extract_origin_header(resolver_headers: Dict[str, Any]):
         return resolved_header[0]
 
     return resolved_header
+
+
+def validate_openapi_security_parameters(
+    security: List[Dict[str, List[str]]],
+    security_schemes: Optional[Dict[str, "SecurityScheme"]],
+) -> bool:
+    """
+    Validates the security parameters based on the provided security schemes.
+
+    Parameters
+    ----------
+    security: List[Dict[str, List[str]]]
+        A list of security requirements
+    security_schemes: Optional[Dict[str, "SecurityScheme"]]
+        A dictionary mapping security scheme names to their corresponding security scheme objects.
+
+    Returns
+    -------
+    bool
+        True if all security scheme names in the `security` parameter are present in the `security_schemes` parameter,
+        False otherwise.
+
+    """
+
+    return bool(
+        security_schemes and all(key in security_schemes for sec in security for key in sec),
+    )
```

Comment:
```suggestion
    
    security_schemes = security_schemes or {}

    security_schema_match = all(
        key in security_schemes for sec in security for key in sec
    )

    return bool(security_schema_match and security_schemes)
```

A few changes:

* **Always guard against a `None`**. Otherwise the `all` will raise `TypeError: argument of type 'NoneType' is not iterable`
* **Break statements when there are two four loops**. It's easy to accidentally get into a [quadratic situation](https://accidentallyquadratic.tumblr.com/). The exercise of breaking it allows you to have another look at the logic to see if there's another way to do it w/o a double loop (not in this case). At worst, it improves readability
* **Use intentional var names on validation**. For schema validation logic, it's easy to get lost when coming back to code as most of them look the same _(e.g., iterating through a data structure)_. Breaking statements, using var names for conditions, or breaking into multiple functions based on conditions make it easier to maintain later.

Q: is there a better naming we can use to make it less ambiguous for these function parameters?

Reply:
>Always guard against a None. Otherwise the all will raise TypeError: argument of type 'NoneType' is not iterable

In this case will not because the first expression test if security_schemes is defined. But your code makes it more clear.

Reply:
DUh, of course, I missed the `and` again.